### PR TITLE
Custom button colors

### PR DESCRIPTION
**Pull request by RubenMattan**

### CODE REVIEW

File: src/rogerthat/bizz/messaging.py [hunk 1]
```diff
@@ -3899,7 +3900,16 @@ def _validate_buttons(buttons, flags, dry_run):
...
+            
+            b.color = MISSING.default(b.color, None)
+            if b.color:
+                b.color = b.color[1:] if b.color.startswith('#') else b.color
+                if len(b.color) == 3:
+                    b.color = b.color[0] * 2 + b.color[1] * 2 + b.color[2] * 2
+                try:
+                    parse_color(b.color)
+                except:
+                    b.color = None
```
Comment (bart-at-mobicage) [inline src/rogerthat/bizz/messaging.py]: Always die --> I would raise a service api exception here

File: src/rogerthat/bizz/service/mfd/__init__.py [hunk 2]
```diff
@@ -708,6 +709,18 @@ def validate_message_flow_definition(mfdef, service_user, from_xml_only_flow=Fal
                            % (answer.caption, printable_id, answer.action,
                               ', '.join(['%s://' % a for a in ALLOWED_BUTTON_ACTIONS])),
                            MessageFlowValidationException)
+                
+            if answer.color:
+                color = answer.color[1:] if answer.color.startswith('#') else answer.color
+                if len(color) == 3:
+                    bizz_check(False, "Button '%s' of Message '%s' has an invalid color '%s', length should be 6."
+                               % (answer.caption, printable_id, answer.color), MessageFlowValidationException)
```
Comment (bart-at-mobicage) [inline src/rogerthat/bizz/service/mfd/__init__.py]: In _validate_buttons color.length == 3 is ok. I'd accept length 3 here as well.

File: src/rogerthat/bizz/service/mfd/__init__.py [hunk 3]
```diff
@@ -708,6 +709,18 @@ def validate_message_flow_definition(mfdef, service_user, from_xml_only_flow=Fal
...
+                try:
+                    parse_color(color)
+                except:
+                    bizz_check(False, "Button '%s' of Message '%s' has an illegal color '%s'."
+                               % (answer.caption, printable_id, answer.color), MessageFlowValidationException)
```
Comment (bart-at-mobicage) [inline src/rogerthat/bizz/service/mfd/__init__.py]: Let's create a validate_color method and re-use it here

File: src/rogerthat/bizz/service/mfd/__init__.py (hunk 2 above)
Comment (RubenMattan) [reply src/rogerthat/bizz/service/mfd/__init__.py]: The clients can't handle length of 3

Comment (bart-at-mobicage) [reply src/rogerthat/bizz/service/mfd/__init__.py]: can we override it here with the 6-char color?

**Pull request merged**